lib/semantics/check-omp-structure.cc

```diff
@@ -88,7 +88,8 @@ void OmpStructureChecker::Leave(const parser::OpenMPLoopConstruct &) {
 }
 
 void OmpStructureChecker::Enter(const parser::OpenMPBlockConstruct &x) {
-  const auto &dir{std::get<parser::OmpBlockDirective>(x.t)};
+  const auto &beginDir{std::get<parser::OmpBeginBlockDirective>(x.t)};
+  const auto &dir{std::get<parser::OmpBlockDirective>(beginDir.t)};
```

Comment:
You may want to check here or in `Leave(const parser::OpenMPBlockConstruct &)` that the `OmpBeginBlockDirective` and `OmpEndBlockDirective` contain the same `Directive`.
The following program is now accepted after the parse-tree changes:

```
real :: a
!$omp parallel
a =1.
!$omp end single
end
```
It maybe part of the tests you intended to do in a later PR, which is fine by me.

Reply:
Thanks! It is already listed in the plan doc (4a). I will do the matching checks along with `LoopConstruct`.